rust/src/data_catalog/unity/models.rs

```diff
@@ -0,0 +1,367 @@
+pub enum TableType {
+    Managed,
+    External,
+    View,
+    MaterializedView,
+    StreamingTable,
+}
+
+///
+#[derive(Deserialize)]
+pub struct TableSummary {
+    /// The full name of the table.
+    pub full_name: String,
+    /// Type of table
+    pub table_type: TableType,
+}
+
+/// A table within a schema
+#[derive(Deserialize)]
+pub struct Table {
+    /// Username of table creator.
+    pub created_by: String,
+
+    /// Name of table, relative to parent schema.
+    pub name: String,
+
+    /// Username of user who last modified the table.
+    pub updated_by: String,
+
+    /// List of schemes whose objects can be referenced without qualification.
+    pub sql_path: String,
+
+    /// Data source format
+    pub data_source_format: DataSourceFormat,
+
+    /// Full name of table, in form of catalog_name.schema_name.table_name
+    pub full_name: String,
+
+    /// Name of parent schema relative to its parent catalog.
+    pub schema_name: String,
+
+    /// Storage root URL for table (for MANAGED, EXTERNAL tables)
+    pub storage_location: String,
+
+    /// Unique identifier of parent metastore.
+    pub metastore_id: String,
+}
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use super::*;
+
+    pub(crate) const ERROR_RESPONSE: &str = r#"
+        {
+            "error_code": "404",
+            "message": "error message"
+        }
+    "#;
+
+    pub(crate) const LIST_SCHEMAS_RESPONSE: &str = r#"
+    {
+        "schemas": [
+            {
+            "created_by": "string",
+            "name": "string",
+            "updated_by": "string",
+            "full_name": "string",
+            "catalog_type": "string",
+            "catalog_name": "string",
+            "storage_root": "string",
+            "storage_location": "string",
+            "properties": {
+                "property1": "string",
+                "property2": "string"
+            },
+            "comment": "string",
+            "created_at": 0,
+            "owner": "string",
+            "updated_at": 0,
+            "metastore_id": "string"
+            }
+        ]
+    }"#;
+
+    pub(crate) const GET_SCHEMA_RESPONSE: &str = r#"
+        {
+            "created_by": "string",
+            "name": "schema_name",
+            "updated_by": "string",
+            "full_name": "catalog_name.schema_name",
+            "catalog_type": "string",
+            "catalog_name": "catalog_name",
+            "storage_root": "string",
+            "storage_location": "string",
+            "properties": {
+                "property1": "string",
+                "property2": "string"
+            },
+            "comment": "string",
+            "created_at": 0,
+            "owner": "string",
+            "updated_at": 0,
+            "metastore_id": "string"
+        }"#;
+
+    pub(crate) const GET_TABLE_RESPONSE: &str = r#"
+    {
+        "created_by": "string",
+        "name": "table_name",
+        "updated_by": "string",
+        "sql_path": "string",
```

Comment:
We could add a test case with omitted `sql_path` but who knows which other fields can be missing from the response under different circumstances...

Reply:
I guess we can just make a bunch of these fields optional. the ones we actually need are quite limited and should always be available - like name etc..

Reply:
WE now just derive default for `Schema` and `Table`

Reply:
I rechecked with the latest changes and deriving Default for the struct does not change the deserializer behavior.
The `#[serde(default)]` macro still needs to be used either per field or on the whole struct making all the fields optional with empty default. An alternative would be to use `Option<String>` where it makes sense.